hadoop-hdds/interface-client/src/main/proto/hdds.proto

```diff
@@ -405,6 +405,7 @@ message ContainerReplicaHistoryProto {
     required int64 firstSeenTime = 2;
     required int64 lastSeenTime = 3;
     required int64 bcsId = 4;
+    required string state = 5;
```

Comment:
An idea was to pull an enum from https://github.com/mladjan-gadzic/ozone/blob/HDDS-7098/hadoop-hdds/interface-server/src/main/proto/ScmServerDatanodeHeartbeatProtocol.proto#L213 and use it here in order not to maintain same enum in two places but I could not make it. Because of that I decided not to use enum at all and stick with string instead.

Reply:
will it be backward compatible if it's a required field, or is backward compat not a concern here?

Reply:
I'd say backward compatibility is not a concern here.

Reply:
It would break rolling upgrade, or when recon and SCM are in different versions.

I see that rolling upgrade is not yet a thing, so I am okay with it.

Reply:
There needs to be a powerful reason for making the proto message `required`. I would change this to optional. If the code depends on it, it should handle it rather than having proto layer throw an exception on parsing.